interfaces/builtin/scsi_generic.go

```diff
@@ -0,0 +1,53 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016-2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+const scsiGenericSummary = `allows access to generic driver (SCSI)devices`
+
+const scsiGenericBaseDeclarationSlots = `
+  scsi-generic:
+    allow-installation:
+      slot-snap-type:
+        - core
+    deny-auto-connection: true
+`
```

Comment:
shouldn't it be as constrained on the plug side as block-devices?

Reply:
but as we know from the use case there's can be almost anything behind these devices, while optical-drive are assumed external. So there can be much larger effects writing  to these. So I think we need to be cautious and assume it's more like block-devices than optical-drive.

Reply:
Plug restriction has been now added.

Reply:
looks correct to me